### PR TITLE
ServerList custom element

### PR DESCRIPTION
**Pull request by strseb**
This pr adds a `<server-list>` custom element. It renders a given serverlist by property and yields events whenever the selected city changes. 

Untitled video - Made with Clipchamp

### CODE REVIEW

**Review by lesleyjanenorton — APPROVED** [review]
This is neat! R+ wc

File: src/components/serverlist.js [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+  }
+
+  const listClasses = { expanded: isCityListVisibile };
+  const listStyles = {
+      height : isCityListVisibile ? serverCountry.cities.length * 48 + "px" : "0px"
+  }
+
+  return html`
+     <li class="server-list-item  ${classMap(listClasses)}" data-country-code="${serverCountry.code}"  @click=${onclick}>
+      <button class="server-city-list-visibility-btn controller ">
```
[the remark this answers is not in the document]
Comment (strseb) [reply src/components/serverlist.js]: Yea, did an audit of unused classes :) remove those.

File: src/components/serverlist.js [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+ * 
+ * @fires ServerList#selectedCityChanged - Fires if selectedCity was changed due to a Click.
+ * 
+ * 
+ */
+export class ServerList extends LitElement {
+  static properties = {
+    serverList: { type: Array },
+    selectedCity: { type: ServerCity },
+    openedCountries: { type: [] },
```
[the remark this answers is not in the document]
Comment (strseb) [reply src/components/serverlist.js]: Great callout, using "opened"

File: src/components/serverlist.js [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+
+  /** @param {ServerCity} city  */
+  #getCityItem(city) {
+    const selectCity = (city) => {
+      if(city == this.selectedCity){
+        return;
+      }
+      this.selectedCity = city;
+      this.dispatchEvent(this.#changeCityEvent(city));
+    }
```
[the remark this answers is not in the document]
Comment (strseb) [reply src/components/serverlist.js]: Sooo i just dropped prettier into the npm:lint task, so we don't need to bother with that. WDYT?

**Pull request merged**